src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -355,23 +401,44 @@ async def sink_buffer(sample: Sample[Quantity]) -> None:
             asyncio.create_task(self._resampler.resample(), name="resample")
         )
 
-    def count_valid(self) -> int:
-        """
-        Count the number of valid samples in this `MovingWindow`.
+    def count_valid(
+        self, since: datetime | None = None, until: datetime | None = None
```

Comment:
Here I think is more important to use keyword-only as both have the same time and could be mixed up.
```suggestion
        self, *, since: datetime | None = None, until: datetime | None = None
```

Reply:
done